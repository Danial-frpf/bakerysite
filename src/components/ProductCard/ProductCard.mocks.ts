import { IProductCard } from ".";

const base: IProductCard = {
    sampleTextProp: "Hello world!",
};

export const mockProductCardProps = {
    base,
};
