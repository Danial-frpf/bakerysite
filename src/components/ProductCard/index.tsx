import Label from "../Label";
import styles from "./styles.module.css";

export interface IProductCard {
    label: string;
    image: string;
    size?: "sm" | "md" | "lg";
    selected?: boolean;
}
const ProductCard: React.FC<IProductCard> = ({
    label,
    image,
    size = "md",
    selected,
}) => {
    let sizeStyle = "";
    switch (size) {
        case "lg":
            sizeStyle = styles.lg;
            break;
        case "md":
            sizeStyle = styles.md;
            break;
        case "sm":
            sizeStyle = styles.sm;
            break;
    }
    return (
        <div
            className={`${styles.container} ${selected ? styles.selected : ""}`}
            data-testid="ProductCard"
        >
            <img
                className={`${styles.image} ${sizeStyle} `}
                src={image}
                alt={`${label} image`}
            />
            <Label label={label} size={size} />
        </div>
    );
};

export default ProductCard;
