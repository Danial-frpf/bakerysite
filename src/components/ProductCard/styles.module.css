.container {
    display: flex;
    background-color: var(--secondary-orange-20);
    align-items: center;
    justify-content: center;
    flex-direction: column;
    padding: 1vh 4vw;
    gap: 2vh;
    width: min-content;
}

.container:hover {
    background-color: var(--secondary-orange-40);
}

.container:active {
    background-color: var(--secondary-blue-40);
}

.selected {
    background-color: var(--secondary-orange-60);
}

.selected:hover {
    background-color: var(--secondary-orange-60);
}

.selected:active {
    background-color: var(--secondary-orange-60);
}

.image {
    max-width: 50vw;
    max-height: 60vh;
    object-fit: contain;
}

.lg {
    max-width: 50vw;
    max-height: 60vh;
}

.md {
    max-width: 40vw;
    max-height: 50vh;
}

.sm {
    max-width: 30vw;
    max-height: 40vh;
}
