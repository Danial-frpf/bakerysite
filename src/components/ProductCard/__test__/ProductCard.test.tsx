import { createRoot } from "react-dom/client";
import { act } from "react-dom/test-utils";
import ProductCard from "..";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);
it("renders correctly without crashing", () => {
    const div = document.createElement("div");
    act(() => {
        createRoot(div!).render(
            <ProductCard label="test" image="/cupcake.png"></ProductCard>
        );
    });
});

it("renders ProductCard correctly", () => {
    const { getByTestId } = render(
        <ProductCard label="test" image="/cupcake.png"></ProductCard>
    );
    expect(getByTestId("ProductCard")).toHaveTextContent("test");
});

it("matches snapshot", () => {
    const tree = renderer
        .create(<ProductCard label="test" image="/cupcake.png"></ProductCard>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
