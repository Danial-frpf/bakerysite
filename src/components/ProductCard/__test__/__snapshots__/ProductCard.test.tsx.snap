// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container "
  data-testid="ProductCard"
>
  <img
    alt="test image"
    className="image md "
    src="/cupcake.png"
  />
  <h4
    data-testid="Label"
  >
    <span>
      test
    </span>
  </h4>
</div>
`;
